### PR TITLE
maintenance api server's default socket should be under root dir

### PR DESCRIPTION
**Pull request by changweige**
We should put all snapshotter's stuff under root dir which can be customized.

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -205,7 +205,11 @@ func SetStartupParameter(startupFlag *command.Args, cfg *Config) error {
 	cfg.GCPeriod = d
 
 	cfg.Address = startupFlag.Address
-	cfg.APISocket = startupFlag.APISocket
+
+	if startupFlag.APISocket == "" {
```
Comment (imeoer) [inline config/config.go:209]: What will happen when `startupFlag.APISocket != ""` ?

File: config/config.go [hunk 2]
```diff
@@ -205,7 +205,11 @@ func SetStartupParameter(startupFlag *command.Args, cfg *Config) error {
...
+		cfg.APISocket = path.Join(cfg.RootDir, "api.sock")
```
Comment (imeoer) [inline config/config.go:210]: It seems we should allow setting the `APISocket` to empty.

Comment (changweige) [reply config/config.go:210]: Do you mean completely shutdown api server?

Comment (imeoer) [reply config/config.go:210]: I mean how do users set it to empty?

**Comment by imeoer** [comment]
It seems the default grpc address is `/run/containerd-nydus/containerd-nydus-grpc.sock`, should we also put the system management api into `/run/containerd-nydus` dir ?

**Comment by changweige** [comment]
> It seems the default grpc address is `/run/containerd-nydus/containerd-nydus-grpc.sock`, should we also put the system management api into `/run/containerd-nydus` dir ?

Yes. It sounds a good idea

File: snapshot/snapshot.go [hunk 3]
```diff
@@ -116,8 +116,8 @@ func NewSnapshotter(ctx context.Context, cfg *config.Config) (snapshots.Snapshot
 		}
 	}()
 
-	if cfg.APISocket != "" {
-		systemController, err := system.NewSystemController(manager, cfg.APISocket)
+	if cfg.EnableSystemController {
+		systemController, err := system.NewSystemController(manager, path.Join(cfg.RootDir, "system_controller.sock"))
```
Comment (imeoer) [inline snapshot/snapshot.go:120]: Maybe name it `supervisor.sock` ?

Comment (changweige) [reply snapshot/snapshot.go:120]: Um. We already have supervisors for nydusd daemon.  Or just rename it to system.sock ?snapshotter.sock ?

Comment (imeoer) [reply snapshot/snapshot.go:120]: `system.sock` is ok.

Comment (changweige) [reply snapshot/snapshot.go]: updated

**Review by imeoer — APPROVED** [review]

**Pull request merged**